### PR TITLE
Issue #1122 layered well from imod5 data

### LINKED ISSUE #1122: Add `from_imod5_data` method to LayeredWell class
https://github.com/Deltares/imod-python/pull/1121 Adds ``LayeredWell`` class, but this still lacks a ``from_imod5_data`` method. This should still be added.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #1122, should be merged after https://github.com/Deltares/imod-python/pull/1121 (so duplication in changeset)

# Description
Adds ``from_imod5_data`` method to ``LayeredWell`` object.

# Checklist
- [x] Links to correct issue
- [ ] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/mf6/wel.py [hunk 1]
```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
...
 
-    def render(self, directory, pkgname, globaltimes, binary):
-        raise NotImplementedError(
-            f"{self.__class__.__name__} is a grid-agnostic package and does not have a render method. To render the package, first convert to a Modflow6 package by calling pkg.to_mf6_pkg()"
-        )
+    y: float or list of floats or np.array of floats
+        is the y location of the well.
+    x: float or list of floats or np.array of floats
+        is the x location of the well.
+    layer: int or list of ints or np.array of ints
```
Comment (luitjansl) [inline imod/mf6/wel.py:756]: is there one layer per well or can there be multiple? If we assign 1 layer as an int,  will it be used for all wells?

File: imod/mf6/wel.py [hunk 2]
```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
...
-                layer, coords=indices_cell2d["mesh2d_nFaces"].coords
-            )
-            face_dim = dst_grid.ugrid.grid.face_dimension
-            indices_cell2d_dims = [face_dim]
-            cell2d_coords = ["cell2d"]
-        else:
-            indices_layer = xr.DataArray(layer, coords=indices_cell2d["x"].coords)
-            indices_cell2d_dims = ["y", "x"]
-            cell2d_coords = ["row", "column"]
+class LayeredWell(GridAgnosticWell):
```
Comment (luitjansl) [inline imod/mf6/wel.py:741]: please add an instance of this class to the package sanity check tests

File: imod/mf6/wel.py [hunk 3]
```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
...
-        # Put dimensions in right order after concatenation.
-        cellid = cellid.transpose("ncellid", "nmax_cellid")
-        # Assign extra coordinate names.
-        coords = {
-            "nmax_cellid": ["layer"] + cell2d_coords,
-            "x": ("ncellid", x),
-            "y": ("ncellid", y),
-        }
-        cellid = cellid.assign_coords(**coords)
+    This package can be written to any provided model grid.
```
Comment (luitjansl) [inline imod/mf6/wel.py:745]: would that be "any model grid that has enough layers"?

File: imod/mf6/wel.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/mf6/wel.py:756]: Scalars are not supported anymore. Docstring had to be updated

File: imod/mf6/wel.py (hunk 2 above)
Comment (JoerivanEngelen) [reply imod/mf6/wel.py:741]: Done

File: imod/mf6/wel.py [hunk 4]
```diff
@@ -410,242 +688,280 @@ def __create_assigned_wells(
...
+        times: list[datetime],
+        minimum_k: float = 0.1,
+        minimum_thickness: float = 1.0,
+    ) -> "Well":
+        pkg_data = imod5_data[key]
+        if "layer" in pkg_data.keys() and (pkg_data["layer"] != 0):
+            log_msg = textwrap.dedent(
+                f"""
+                In well {key} a layer was assigned, but this is not
+                supported. Assignment will be done based on filter_top and
```
[the remark this answers is not in the document]
Comment (JoerivanEngelen) [reply imod/mf6/wel.py:705]: Extended error message

File: imod/mf6/wel.py [hunk 5]
```diff
@@ -376,27 +654,27 @@ def _validate(self, schemata: dict, **kwargs) -> dict[str, list[ValidationError]
...
+    def _assign_wells_to_layers(
         self,
         wells_df: pd.DataFrame,
         active: GridDataArray,
         top: GridDataArray,
         bottom: GridDataArray,
         k: GridDataArray,
-        minimum_k: float,
-        minimum_thickness: float,
     ):
```
[the remark this answers is not in the document]
Comment (JoerivanEngelen) [reply imod/mf6/wel.py:664]: Added return type

File: imod/mf6/wel.py [hunk 6]
```diff
@@ -70,7 +71,307 @@ def mask_2D(package: Well, domain_2d: GridDataArray) -> Well:
...
+
+        # Groupby index and select first, to unset any duplicate records
+        # introduced by the multi-indexed "time" dimension.
+        df_for_cellid = wells_assigned.groupby("index").first()
+        d_for_cellid = df_for_cellid[["x", "y", "layer"]].to_dict("list")
+
+        return self._derive_cellid_from_points(like, **d_for_cellid)
+
+    def _create_dataset_vars(
+        self, wells_assigned: pd.DataFrame, wells_df: pd.DataFrame, cellid: xr.DataArray
```
[the remark this answers is not in the document]
Comment (JoerivanEngelen) [reply imod/mf6/wel.py:126]: Good one, I realized that the ``wells_df`` dataframe didn't need to be passed through, so removed that. Index names can also be retrieved from ``wells_assigned``.

File: imod/mf6/model.py [hunk 7]
```diff
@@ -267,7 +268,7 @@ def write(
         mf6_hfb_ls: List[HorizontalFlowBarrierBase] = []
         for pkg_name, pkg in self.items():
             try:
-                if isinstance(pkg, imod.mf6.Well):
+                if isinstance(pkg, GridAgnosticWell):
```
[the remark this answers is not in the document]
Comment (JoerivanEngelen) [reply imod/mf6/model.py:271]: Done!

**Pull request merged**